### PR TITLE
Adding packages/rosetta

### LINKED ISSUE #4694: Write .ros files for Rosetta

### PR DESCRIPTION
**Pull request by joeabbey**
closes: #4694

## Description

Introduces a "packages/rosetta" to run rosetta-cli.  A docker-compose.yaml file is provided to simplify the orchestration of 4 services (agoric, rosetta, a faucet, and test).

The `agoric` service is effectively `cosmic-swingset-solo:latest` with some additional changes to run `scenario2` with a INITIAL_HEIGHT of 1.

The `rosetta` service runs `agd rosetta`, with additional config, to connect to the agoric service and expose the rosetta API.  This can be folded into the agoric service in a future PR.  This service waits to connect until the agoric service begins producing blocks.

The `faucet` service allows rosetta-cli to send 10 $BLD and 10 $IST as needed to a new address.  This service doesn't wait, but won't work until `agoric` service has begun producing blocks.  It relies on the `bootstrap` address in the keyring to send funds.

The `test` service waits for the rosetta service to be ready.  Once it is ready, the service executes rosetta-cli to perform the data check and construction check.  This tests for general compliance with Rosetta's API, as well as demonstrating account creation and transfers of $BLD and $IST.  This container will exit with zero upon success.

Build the `rosetta-ci:latest` container with:

```
cd <agoric-sdk>
./scripts/build-rosetta-ci.sh
```

Run the tests with:

```
cd <agoric-sdk>/packages/rosetta
docker-compose up --abort-on-container-exit --exit-code-from test
```

### Security Considerations

None.

### Documentation Considerations

The cosmos-sdk has additional `app.toml` configuration (or command line options) to enable rosetta.  For instance, to enable for mainnet:

```toml
[rosetta]

# Enable defines if the Rosetta API server should be enabled.
enable = true

# Address defines the Rosetta API server to listen on.
address = ":8080"

# Network defines the name of the blockchain that will be returned by Rosetta.
blockchain = "app"

# Network defines the name of the network that will be returned by Rosetta.
network = "agoric-3"

# Retries defines the number of retries when connecting to the node before failing.
retries = 3

# Offline defines if Rosetta server should run in offline mode.
offline = false
```

### Testing Considerations

Either in this PR or in a future PR, this should be integrated into a `.github/workflows` to ensure that compatibility is maintained as both the cosmos-sdk and rosetta-sdk evolve over time.  Whether or not `docker-compose` is used for this CI job is certainly up for discussion.

### CODE REVIEW

File: packages/rosetta/configuration/transfer.ros
Comment (michaelfig) [inline packages/rosetta/configuration/transfer.ros:3]: Should this instead use 6 decimals, since this is micro-BLD?  I'm not sure of the implications.

```suggestion
    currency = {"symbol":"ubld", "decimals":6};
```

File: packages/rosetta/configuration/transfer.ros
Comment (michaelfig) [inline packages/rosetta/configuration/transfer.ros:109]: Again here.

**Review by michaelfig — COMMENTED** [review]
Looks good!  I'll wait for more folks to have a look before I can approve.

File: packages/rosetta/configuration/transfer.ros
Comment (joeabbey) [reply packages/rosetta/configuration/transfer.ros:3]: I thought so too... but I think that the Rosetta "currency/amount decimals" are not the same as Cosmos SDK's "denom decimals":

https://www.rosetta-api.org/docs/models/Currency.html
https://www.rosetta-api.org/docs/models/Amount.html

Given the example:

```
{
    "symbol": "BTC",
    "decimals": 8,
}
```
and `100000000` as a value means 1 BTC.  If we apply it to BLD... then BLD should be:

```
{
    "symbol": "BLD",
    "decimals": 6
}
```

such that `1000000` as a value means 1 BLD.

If we use 6 decimals for ubld here:

```
{
    "symbol": "ubld",
    "decimals": 6
}
```

it means that 1000000 as a value is not 1 BLD (to Rosetta) but is 1 ubld!

On the other hand... if we use:

```
{
    "symbol": "ubld",
    "decimals": 0
}
```

This means that 1000000 as a value is 1000000 ubld.  Presumably, Rosetta consumers will then have a ubld to BLD conversion to adjust from 1000000 ubld to 1BLD.

File: packages/rosetta/configuration/transfer.ros
Comment (joeabbey) [reply packages/rosetta/configuration/transfer.ros:109]: See above 🙏

File: packages/rosetta/configuration/bootstrap.json
Comment (JimLarson) [inline packages/rosetta/configuration/bootstrap.json]: Add newline.

File: packages/rosetta/configuration/rosetta.json
Comment (JimLarson) [inline packages/rosetta/configuration/rosetta.json]: Add newline at end of file.

File: packages/rosetta/rosetta-ci/Dockerfile
Comment (JimLarson) [inline packages/rosetta/rosetta-ci/Dockerfile]: Add newline at end of file.

File: scripts/build-rosetta-ci.sh
Comment (JimLarson) [inline scripts/build-rosetta-ci.sh]: Add newline at end of file.

File: packages/rosetta/configuration/run_tests.sh
Comment (JimLarson) [inline packages/rosetta/configuration/run_tests.sh]: FYI, timeout is not a standard UNIX command, but this is okay if this script is only run under Docker.

File: packages/rosetta/configuration/send_funds.sh
Comment (JimLarson) [inline packages/rosetta/configuration/send_funds.sh:7]: Note that `ADDR` must be set in the environment.  Did you mean to use `DESTINATION` where you use `ADDR` below? The invocation in `faucet.py` does not seem to set `ADDR`.

**Review by JimLarson — CHANGES_REQUESTED** [review]

File: packages/rosetta/configuration/send_funds.sh
Comment (joeabbey) [reply packages/rosetta/configuration/send_funds.sh:7]: whoops.. they were only used in output messages and in the filenames... which happened to be luckily unimportant... fixing

File: packages/rosetta/configuration/run_tests.sh
Comment (joeabbey) [reply packages/rosetta/configuration/run_tests.sh]: Rewrote as a bash while loop..

**Review by JimLarson — APPROVED** [review]

**Review by arirubinstein — APPROVED** [review]

**Comment by michaelfig** [comment]
@Mergifyio help

**Comment by michaelfig** [comment]
@Mergifyio refresh

**Comment by michaelfig** [comment]
@Mergifyio update

**Pull request merged**
